### DISCUSSION TITLE
Failed to open permalink: parsing Git remote

### DISCUSSION THREAD

**Discussion by mfridman** [zed-industries/zed#10215] [Feature Requests]
Maybe this is a bug, or my misunderstanding on what this option is.

Zed 0.129.2

I highlighted a few lines of code and was hoping to get the GitHub permalink, like this:

https://github.com/golang-jwt/jwt/blob/c8043eab61f0ec5bdd924c1c30caf164a9bb2c66/hmac.go#L9-L14

![CleanShot 2024-04-05 at 18 57 14@2x](https://github.com/zed-industries/zed/assets/6278244/0899f641-0880-441a-98dd-7d7446e2dc5f)

But instead, I see a parse error:

![CleanShot 2024-04-05 at 18 54 56@2x](https://github.com/zed-industries/zed/assets/6278244/3fbcfeb7-e669-4cb9-85c6-cff7779c42d8)

```sh
git remote -v
origin	ssh://git@github.com/golang-jwt/jwt (fetch)
origin	ssh://git@github.com/golang-jwt/jwt (push)
```

Am I doing something wrong, or should this handle `ssh://` in addition to `https://`?

**Comment by slocoro** [comment]
I'm getting the same error on `Zed 0.148.0`

**Comment by takayukioda** [comment]
It [seems](https://github.com/zed-industries/zed/blob/b578be5c77c3fe11f4860e569e5558526d520288/crates/git_hosting_providers/src/providers/github.rs#L112-L125) it's currently support only the remotes start with `git@github.com:` or `https://github.com/`

I checked [the git document](https://git-scm.com/docs/git-clone/en#_git_urls) and it says git treats `<user>@<host>:` as a ssh protocol, so I guess that's why.
It's not friendly, but at least we now know how to fix in either zed code or in our settings.

**Comment by notpeter** [comment]
I believe ssh urls are fixed with:
- https://github.com/zed-industries/zed/pull/19924

**Comment by mdpstory** [comment]
i also get this error "Failed to open permalink: parsing Git remote", because i have gitlab installed on premise and i clone it with ip example: http://192.168.0.123

**Reply by notpeter** [reply]
Zed does not currently support custom https base_urls configured from git remotes.

But...if you have a hostname that includes `gitlab` Zed will use proper pathing for Gitlab self-hosted permalinks.

See: [Gitlab::from_remote_url](https://github.com/zed-industries/zed/blob/ecaf44511cd1a1efc6ded4b56f87f1435dbc0e07/extensions/zed/crates/git_hosting_providers/src/providers/gitlab.rs#L26-L49) in the source.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
